src/relations/mysql_tls.py

```diff
@@ -118,12 +131,16 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
 
     def _on_tls_relation_broken(self, _) -> None:
         """Disable TLS when TLS relation broken."""
-        self.charm.set_secret(SCOPE, "ca", None)
-        self.charm.set_secret(SCOPE, "cert", None)
-        self.charm.set_secret(SCOPE, "chain", None)
+        try:
+            self.charm.set_secret(SCOPE, "ca", None)
+            self.charm.set_secret(SCOPE, "cert", None)
+            self.charm.set_secret(SCOPE, "chain", None)
+        except KeyError:
+            # ignore key error for unit teardown
+            pass
         self.remove_tls_config_file()
         # set member-state to avoid unwanted health-check actions
-        self.charm.unit_peer_data["member-state"] = "waiting"
+        self.charm.unit_peer_data.update({"member-state": "waiting", "tls": ""})
```

Comment:
should we be deleting the `tls` key from the unit_peer_data instead?

Reply:
setting it to empty string (or None) will delete it from data bag